src/BSON/Binary.c

```diff
@@ -387,6 +388,26 @@ phongo_create_object_retval php_phongo_binary_create_object(zend_class_entry *cl
 #endif
 } /* }}} */
 
+static int php_phongo_binary_compare_objects(zval *o1, zval *o2 TSRMLS_DC) /* {{{ */
+{
+	php_phongo_binary_t *intern1, *intern2;
+
+	intern1 = Z_BINARY_OBJ_P(o1);
+	intern2 = Z_BINARY_OBJ_P(o2);
+
+	/* MongoDB compares binary types first by the data length, then by the type
+	 * byte, and finally by the binary data itself. */
```

Comment:
I expect HHVM won't be able to implement this. See: [PHPC-654](https://jira.mongodb.org/browse/PHPC-654?focusedCommentId=1443092&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-1443092).